astropy/coordinates/matching.py

```diff
@@ -457,7 +457,7 @@ def _get_cartesian_kdtree(coord, attrname_or_kdt='kdtree', forceunit=None):
         else:
             cartxyz = coord.cartesian.xyz.to(forceunit)
         flatxyz = cartxyz.reshape((3, np.prod(cartxyz.shape) // 3))
-        kdt = KDTree(flatxyz.value.T)
+        kdt = KDTree(flatxyz.value.T, compact_nodes=False, balanced_tree=False)
```

Comment:
This change looks good, but will only work for the Cython implementation of KDTree, correct? I don't think the pure python version accepts these arguments.

You'll either need to add some logic to use different arguments if the Python implementation is being used, or (better IMO) put this in a try-except block which drops down to the old call in the case of a ```TypeError````